### ISSUE TITLE
Filter out Yahoo Referrer params tags

### ISSUE THREAD

**Issue by ryanbr** [brave/brave-browser#25691]
We should filter out guccounter,guce_referrer, & guce_referrer_sig

Already blocked by default in Adguard remove params;
https://github.com/AdguardTeam/AdguardFilters/blob/master/TrackParamFilter/sections/general_url.txt#L188

```
$removeparam=guccounter
$removeparam=guce_referrer
$removeparam=guce_referrer_sig
```

Clean url; https://github.com/brave/adblock-lists/pull/978

Example urls: 

```
https://www.yahoo.com/news/onlinecheckwriter-launches-cloud-based-check-031500636.html?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuemlsbW9uZXkuY29tLw&guce_referrer_sig=AQAAALtlTSPpvxhF3tU3Xj9MHxAREbMjIDHCnJrOfdbk5IOa7spOcvIboX5nUui4__vVfPBRax4kVk2t-8PYSBqci5bE_pq3-3kirOriIgR2-mgkMb3bX6MRt3GdKLteyh4QnaJG5ORbcd3Uod1YQ7L_ixl0EeFB90yxlw_66o5g51Jd

https://www.engadget.com/?guccounter=1
https://de.yahoo.com/?p=us&guccounter=1
```

**Comment by fmarier** [comment]
@ryanbr Have you seen any examples where `guccounter` is set to another value than `1`?

Are the other parameters used cross-site? It looks like it's just something internal to Verizon properties to [keep track of whether or not people have consented to targeted ads](https://old.reddit.com/r/yahoo/comments/czju1g/guce/ez945sl/). Unless we see evidence to the contrary, it looks to me like it's some first-party URL decoration that's out-of-scope in the query filter, but is definitely in-scope for the copy clean URL feature.

**Comment by ryanbr** [comment]
I haven't see cases where value wasn't set to other than `1`. The patch was just syncing with existing fixes in Adguard who are also filtering the same params without issue.  Whether those params is actually even needed to view a page also, GUCE will set cookies for the consents to show. @fmarier

**Comment by fmarier** [comment]
Here's a bunch of examples from various third-party sites:
https://ca.finance.yahoo.com/news/stage-32-changing-way-hollywood-104853891.html?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuc3RhZ2UzMi5jb20v&guce_referrer_sig=AQAAAJ2RmkcSJn2Ov4LUhMul2DCW3fyaDQe91PtOq-2CeNTgwy6APa0uiX0wqJ7Yw3ICi5hCvj4gD-avlH73qgf9_ob0r3rVs2rXn1OYo-J1Y8nqsW6ykscFEV-5_qSagm74Lo_x8WUTHgtDJtVsMyxlfV57kDfCBiuYjCkPcX5R_Ad7
https://finance.yahoo.com/news/tell-tinder-match-scamming-193648255.html?guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmRrLw&guce_referrer_sig=AQAAAMfqnNhM2kLjKfPgXzIEXstdhh83StBGs2fLFKOVARIPMs64LV2ja_9lDd8NZdGin5n46bC8BLjq9zRGLuuJomfzzhA-hoaF_FZCQ
https://www.yahoo.com/news/onlinecheckwriter-launches-cloud-based-check-031500636.html?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuemlsbW9uZXkuY29tLw&guce_referrer_sig=AQAAALtlTSPpvxhF3tU3Xj9MHxAREbMjIDHCnJrOfdbk5IOa7spOcvIboX5nUui4__vVfPBRax4kVk2t-8PYSBqci5bE_pq3-3kirOriIgR2-mgkMb3bX6MRt3GdKLteyh4QnaJG5ORbcd3Uod1YQ7L_ixl0EeFB90yxlw_66o5g51Jd
https://sports.yahoo.com/news/vida-cannabis-corp-vida-cannabis-154201658.html?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAAAMLvsfLdqXEy2zK3iBEMa-vKSaM3WWxYgAnXHKCDwiqrZ75o3OYFg1uMwK1HaSVuNLAg7NpSTsbNhB4dVAAdRniXKknWOI3WkkcBgnlQis8bcUa4-bISjgUFMveS44zI8tM98da5dBu-RkMAtm9jrpFaxWCZ8lz5gjgwXIzreIs
https://es-us.noticias.yahoo.com/bolivia-dirigir%C3%A1-fondo-desarrollo-pueblos-000348935.html?guce_referrer=aHR0cHM6Ly9hcHAubWVsdHdhdGVyLmNvbS8&guce_referrer_sig=AQAAABBpV6Png58vSHjZjdAbzSHl7Q1Nu_bBugsCvrfl5zRd0HYwwyYLjk5hzCdMjvssfYtVMw-vhrePRTSH2JUYLgFyFXyJyNtj18DDEqEVSBLVoM0KW2JmNDI68Mp9Lazvcj_CxZs669NqI5kYYMxIPbNJ6F2BsTZlrhVhuW1EDhzM&guccounter=2
https://es-us.noticias.yahoo.com/pa%C3%ADses-iberoamericanos-reclaman-acceso-vacunas-093138429.html?guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAAMlZhrJzqc7hYlOCr6_TrgFa4fdpp4PhminOZAIFYdRIx_pE0OlYpEA7SXI5cqd5ZFT9bOf46mxf4UGmtdOnVPrWXkOwnpUEFzu-z81NhQLA5MfcbknWKwNQFysAo8iYSUuv_81SxWWCcZDzwLCriPYhKRSoc3ggSTPeaQL4KrWb&guccounter=2
https://finance.yahoo.com/news/most-affordable-college-state-130050517.html?guccounter=1&guce_referrer=aHR0cHM6Ly9hcHAubWVsdHdhdGVyLmNvbS8&guce_referrer_sig=AQAAAF09JCe4kxxKYkhvzrhERCPnfzEvDg_Hj2TimMLvDI2AGgZQhzfZRjM1C_FpNtQskh3peJumnZPeL5imFq645MbePU4IK2l5SO_QxV-z3pNn2r0Ti6pga0yfPMF2AYMyroUtdvlir8fxLPxanQq_E75vLy7m92NHtIuSMoCz_edi
https://www.huffpost.com/entry/grid110-how-a-group-of-di_b_9704202?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ3JpZDExMC5vcmcvcHJlc3M&guce_referrer_sig=AQAAALWrx0tF5V25WHcEpsy_jWZUwPfY51hjb1uT4kIxB81MYrK52vx9HgNAkWmpjXXVBCp0nx7bJ9lUDwdZQUni5jNRB90cT145-g2_J1Y6DiG0Xv8JXrkGdyr3iw69lKEtbLSb0JKUvRNWF5FkmPk_PI6EA38_vqVw4rSpnMtpPd1V
https://tw.news.yahoo.com/%E5%A4%A7%E6%B9%96%E6%8E%A1%E8%8D%89%E8%8E%93-%E9%AB%94%E9%A9%97%E6%8E%A1%E6%9E%9C%E6%A8%82-160000510.html?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAAGieWHTCkWFio4G2-Hp5OmSZpZ8UmTdFqrLAZv12_b
https://techcrunch.com/?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAABaaf8fyonwBu6H8lVmgrljU_vU0Cw8EenxoBsG3lE3nmqwPre5FoHLa29vhPW6gtSCXuQp2QBsGxxgQ9RWdGKNuFHIspqdJ3R2mvFTX_aJHkFN0ptxIH65h9JwfUMyDhm-s_gKbeoN4A5CVLcqzRzOCm-_UoMT9a40AsiRmgcnt
https://techcrunch.com/2016/11/09/sentione-raises-3-5-million-to-monitor-social-media/?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAAHu0Au5mG7Hcjz9FN4r-5p0qUT6UR_hWI0rmc6c5xf5UqjPyI0eE_zNAtCOrFjlb03lDP5vUSsQhi3wFM0e5rte9qP6ZEwi4KEz8sd9UGdsMRiDEbd9QhvOK692Ie5mlLhXrw2P4D6xwpqY3FWb1DDHmFnwIg00-METLCZZRSOXd
https://news.yahoo.com/dr-kojian-helping-millions-americans-063621440.html?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZHJ0b2hlbHAuY29tLw&guce_referrer_sig=AQAAAByI8ZFmKw6ep8IXdP8IM6-rjzPAheddb0NirD2gtZBS5A2ch3GOqKbgyUNQwjP-xx3e40M6Dq4tZUEgMO0rUrEK5A2gYaiy7ejXIl0OnFTAYDKm_5FdUWqmwvm7nE-I1wzLGFEhWLxMLeFkTpcreQSeF9A4-VD_DSQilnkw_LnI
https://www.yahoo.com/now/chups-unveils-first-online-marketplace-180000664.html?guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAAGy9GYJttIXLvg4ex0wR81yt7qvH6UEFvwDcA8XWKXLZiuEioQZd1B3sDmkb41Zkszb3evKHf_Zz6UPlSdWfy2OhD2PNZWKst7ggzelZ-MU3kQbkgl3iMLgUJMGP5md02c4IKepE0Eeds9lxi1U7T9MGlzcWSMNtVsvSNGyH928p&guccounter=2
https://www.huffingtonpost.co.uk/entry/rich-kids-of-instagram-charity-work-hope-for-children_uk_58ede073e4b0ca64d91a6449?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuc2FtdWVsYW5kY290cmFkaW5nLmNvbS8&guce_referrer_sig=AQAAAIVsAEjplGIykuPe2fyk2cELwfBu-u8jrJ9iBEg5wicq358ea-qz3zsAv5ArVcHjnsSB4C5RFoAdcrh2WORu4RltP_Ks22l3Y1MkxTYDmdtrW9elPQsf2RfJaTXzjX2ioISezToZZObpxjy-7d8o_LdzKGyOMBZs2ecTLvEMlbG6
https://www.huffingtonpost.co.uk/entry/killing-eve-sandra-oh-queerbaiting_uk_5cf9087ae4b0e3e3df160b9b?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAAB4cAdLMRRJhTRnjhGE6SV_QTFtvnJc3QSiW_Fn0pSNM94gb3catOC4YRWyAKodzubO5PQQV4beekbuAPQJT61V1fwdpEtp6qC7gXfV9QhhdkzeKmMmVFNC7fa3FeZ1AHlhf863hGPxv0YpiLF3C5xnJ0c4kt_P29_v0dYSDk3gM
https://www.yahoo.com/lifestyle/thrive-coworking-introduces-philanthropic-community-133000733.html?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAAKN_K2rG_1PEM5P6bvAnClsrub0Oz8k31nU2E00arEOlHaqElpn8xySXS_9_f9Jyq51YIEEg-9_NlZADUUJAlA-p2jOyCP4irmyYR_PJ3TQQCKhxgaEpN6iS6rwd18M4kjZisVn82P5ArXcz94215VHAfh4PH2wcFrbVIgq43q0S
https://techcrunch.com/2022/06/30/how-to-keep-your-development-team-aligned-with-the-companys-product-vision/?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAALn4Mej4lpcets3JBEPaWoQFWkDFRp00rmepNOe8bEN4ArTor_IpiXuakPjQCdBMojMwVErlf1Kcr0G_-5Vvm2aQpYD96C-t-8MUS9wlBWswvnlmbmylujK8IxEHXpqzqYK6bL4R0sIH82FfJ9ZoXo4L6qCdFQQcAea0PMzlS9wV
https://finance.yahoo.com/news/neuroxess-attends-waic-first-medical-120000689.html?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAAAT0QWgTNKpZRapBldRoEze4LGbv_Mr4ktO292bD1ZiFL0we7qtqJ4OS_SEnjTXfD5J8PC4LDBRa8hWQzWwa-DiPfH6
https://www.huffpost.com/entry/zane-carney-the-future-of-electric-guitar_b_5a159f08e4b0f401dfa7ec7d?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuZ29vZ2xlLmNvbS8&guce_referrer_sig=AQAAAKdEDeS_ViGCPjZrmPL9WgDLQCiu6qtfrrsi0jUxJkYQkfPH9GNES6Mxtvelm0Odbn8TEbjknob8Sn0dJIC7mnqQ80I5yjpOBd3fFZgelMOUs-yL4aWlbU-VOo-dlG2syh1Rvsj-N40dxoCsr93x6zKgffPCK712JtyRnX270Yle

It's not clear exactly what this is tracking, but according to https://old.reddit.com/r/yahoo/comments/czju1g/guce/ez945sl/, it seems to only affect the type of ads that users will see on these properties. So I agree, we should remove `guce_referrer` and `guce_referrer_sig`.

`guccounter=1` on the other hand is not a tracker since it's always set to 1 and therefore [out-of-scope for the default-on list](https://github.com/brave/brave-browser/wiki/Query-String-Filter).

**Comment by stephendonner** [comment]
## Verified `PASSED` using 

Brave | 1.48.118 Chromium: 109.0.5414.80 (Official Build) beta (x86_64)
-- | --
Revision | 0f69b168d36a06cace4365e9f029fa987afa5633-refs/branch-heads/5414@{#1178}
OS | macOS Version 11.7.2 (Build 20G1020)

### Steps:
1. installed `1.48.118`
2. launched Brave
3. loaded `https://brave.com/?guce_referrer=1234&guce_referrer_sig=abcd&ref_src=foobar`
4. examined the URL bar and `Developer Tools` -> `Network` for the original request

#### Confirmed the gucci_ parameters were stripped

<img width="1763" alt="Screen Shot 2023-01-09 at 4 47 16 PM" src="https://user-images.githubusercontent.com/387249/211437291-5f16a378-2f17-47a2-95c5-360d64762598.png">

**Comment by stephendonner** [comment]
## Verified `PASSED` using `1.48.134` on a `Google Pixel XL` running `Android 9`

### Steps:

1. installed `1.48.134`
2. launched Brave
3. connected via `brave://inspect/#devices`
4. loaded `https://brave.com/?guce_referrer=1234&guce_referrer_sig=abcd&ref_src=foobar`
5. examined the URL bar and network headers


#### Confirmed the `guce_*` parameters were stripped, and the final URL was `https://brave.com/?ref_src=foobar`

`brave://inspect/#devices` | `Google Pixel XL`
----------------------------|------------------
<img width="1578" alt="Screen Shot 2023-01-21 at 12 15 19 AM" src="https://user-images.githubusercontent.com/387249/213858609-57d8c44c-7074-493c-a791-29d1a71945ef.png"> | ![Screenshot_20230121-001507](https://user-images.githubusercontent.com/387249/213858597-5c4f6c1b-e379-4b3b-b085-c3a993498a28.png)
